app/javascript/hooks/mutations/sessions/useCreateSession.jsx

```diff
@@ -0,0 +1,26 @@
+import {useMutation, useQueryClient} from "react-query";
+import axios, { ENDPOINTS } from "../../../helpers/Axios";
+import {useNavigate} from "react-router";
+
+const createSession = (sessionUser) => axios.post(ENDPOINTS.signin, sessionUser)
+  .then((resp) => resp.data)
+  .catch((error) => console.log(error));
+
+export function useCreateSession(options){
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+
+  const mutation = useMutation(createSession, {
+    // Re-fetch the current_user and redirect to homepage if Mutation is successful.
+    onSuccess: () => {
+      queryClient.invalidateQueries('useSessions');
+      navigate('/rooms');
+    },
+    onError: (error) => {
+      console.log('mutate error', error);
+    },
+  });
+
+  const onSubmit = (user_data) => mutation.mutateAsync({ user: user_data }).catch(/*Prevents the promise exception from bubbling*/()=>{})
```

Comment:
Is the `onSubmit` defined but never been used?
We can rename it as `handle[Subject] (handleSubmit)` and use it when needed i.e. in the `app/javascript/components/sessions/SignIn.jsx`